### DISCUSSION TITLE
路径解析的问题

### DISCUSSION THREAD

**Discussion by henryzp** [alibaba/formily#2393] [Q&A 2.x] [answered]
写了一个Field组件：

```jsx
/** Field使用的是@formily/antd的组件 */
<Field
  name="{code:a,name:b}"
  component={[
    SimpleSelect,
    {
      placeholder: '请选择业务类型',
      selectLabel: 'name',
      fetchUrl: `/gm1-plus-rest/plus/booking/getBusiType?tab=0&busiType=1`,
      useFormily: true,
    },
  ]}
/>
```

我的SimpleSelect组件，最终onChange事件的第一个参数大概是长这样的：{code: '123', name: 'hello'}

期望formily能帮助我将code值赋于a字段，name赋于b字段，即得到{a: '123', b: 'hello'}的结果

但实际效果是页面会报错

![image](https://user-images.githubusercontent.com/3849702/140071764-1f324375-a2e7-4a3d-a873-d313376c2ff2.png)

是我哪里使用不对么？

**Comment by janryWang** [comment, marked as answer]
这是react报错，如果a,b不存在的时候，会默认给组件塞个空对象，应该是你的组件接收到了空对象，直接就渲染了

**Reply by henryzp** [reply]
没看明白。。。那这个要怎么破？一开始在form里面就要给a和b一个初始值么？

**Reply by janryWang** [reply]
这是你组件自己的报错，跟formily没关系，自己排查下就知道了

**Reply by henryzp** [reply]
已经找到原因了，感谢
